### PR TITLE
Add floordata to rooms window filters

### LINKED ISSUE #1223: Add floordata to rooms window filters
Allow filters to operate on floordata types present in sectors.

### PR DESCRIPTION
**Pull request by chreden**
Allow filters to operate on floordata types present in sectors.
Closes #1223

### CODE REVIEW

File: trview.app/Elements/Floordata.cpp [hunk 1]
```diff
@@ -326,4 +326,54 @@ namespace trview
...
+            case Floordata::Command::Function::ClimbableWall:
+                return "Climbable Wall";
+            case Floordata::Command::Function::Triangulation_Floor_NWSE:
+                return "Triangulation Floor NWSE";
+            case Floordata::Command::Function::Triangulation_Floor_NESW:
+                return "Triangulation Floor NESW";
+            case Floordata::Command::Function::Triangulation_Ceiling_NW:
+                return "Triangulation Ceiling NW";
+            case Floordata::Command::Function::Triangulation_Ceiling_NE:
+                return "Triangulation_Ceiling_NE";
```
Comment (lahm86) [inline trview.app/Elements/Floordata.cpp:353]: Maybe the underscores throughout aren't needed for the descriptions.

File: trview.app/Elements/Floordata.cpp [hunk 2]
```diff
@@ -326,4 +326,54 @@ namespace trview
...
+            case Floordata::Command::Function::Triangulation_Ceiling_Collision_SW:
+                return "Triangulation Ceiling Collision_SW";
+            case Floordata::Command::Function::Triangulation_Ceiling_Collision_NE:
+                return "Triangulation Ceiling Collision_NE";
+            case Floordata::Command::Function::Triangulation_Ceiling_Collision_NW:
+                return "Triangulation Ceiling Collision_NW";
+            case Floordata::Command::Function::Triangulation_Ceiling_Collision_SE:
+                return "Triangulation Ceiling Collision_SE";
+            case Floordata::Command::Function::MonkeySwing:
+                return "MonkeySwing";
```
Comment (lahm86) [inline trview.app/Elements/Floordata.cpp:371]: Maybe a space here.

File: trview.app/Elements/Floordata.cpp [hunk 3]
```diff
@@ -326,4 +326,54 @@ namespace trview
...
+            case Floordata::Command::Function::MinecartLeft_DeferredTrigger:
+                return "MinecartLeft DeferredTrigger";
+            case Floordata::Command::Function::MinecartRight_Mapper:
+                return "MinecartRight Mapper";
```
Comment (lahm86) [inline trview.app/Elements/Floordata.cpp:375]: Maybe spaces and slashes here like "Minecart left / Deferred trigger"?

**Review by lahm86 — COMMENTED** [review]
Looks great, just some wordy things.

I don't know if it's feasible to repurpose the descriptions that show in the FD tab rather than have the switch case here?

image

**Comment by chreden** [comment]
> I don't know if it's feasible to repurpose the descriptions that show in the FD tab rather than have the switch case here?

Hmm yes I suppose so if I add the triangulation directions to the FD tab, will do that and call new function from there.

**Review by lahm86 — APPROVED** [review]

**Pull request merged**